dbms/src/DataStreams/WindowTransformAction.cpp

```diff
@@ -1294,10 +1297,58 @@ void WindowTransformAction::writeOutCurrentRow()
     }
 }
 
+RowNumber WindowTransformAction::writeBatchResult()
+{
+    assert(partition_ended);
+    assert(current_row < partition_end);
+
+    size_t insert_row_num = 0;
+    const auto & block = blockAt(current_row);
+    RowNumber tmp_row;
+    if (current_row.block != partition_end.block)
+    {
+        insert_row_num = block.rows - current_row.row;
+        tmp_row = current_row;
+        tmp_row.block += 1;
+        tmp_row.row = 0;
+    }
+    else
+    {
+        insert_row_num = partition_end.row - current_row.row;
+        tmp_row = partition_end;
+    }
+
+    // As aggregation function and window function can not occur in one window at the same time
+    // Only batch inserting results for aggregation function is ok
+    for (auto & ws : aggregation_workspaces)
+    {
+        IColumn * result_column = block.output_columns[ws.idx].get();
+        const auto * agg_func = ws.aggregate_function.get();
+        auto * buf = ws.aggregate_function_state.data();
+        agg_func->batchInsertSameResultInto(buf, *result_column, insert_row_num, nullptr);
```

Comment:
since the last argument is always nullptr, why add the last argument in `batchInsertSameResultInto`

Reply:
> since the last argument is always nullptr, why add the last argument in `batchInsertSameResultInto`

deleted